src/Status.hpp

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include "utils.h"
+
+namespace FileWriter {
+class Streamer;
+namespace Status {
+
+class StdIOWriter;
+class JSONWriter;
+class FlatbuffersWriter;
+
+class StreamerStatus;
+class StreamMasterStatus;
+
+// Error codes for the StreamMaster and Streamers
+enum StreamMasterErrorCode {
+  not_started = 0,
+  running = 1,
+  has_finished = 2,
+  streamer_error = -1,
+  statistics_failure = -10,
+  streammaster_error = -1000
+};
+enum StreamerErrorCode {
+  no_error = 1001,
+  writing = 1,
+  stopped = 0,
+  configuration_error = -1,
+  consumer_error = -2,
+  metadata_error = -3,
+  topic_partition_error = -4,
+  assign_error = -5,
+  topic_error = -6,
+  offset_error = -7,
+  start_time_error = -8,
+  message_error = -9,
+  write_error = -10,
+  not_initialized = -1000,
+};
+
+const std::string Err2Str(const FileWriter::StreamMasterError &);
+const std::string Err2Str(const FileWriter::StreamerError &);
+
+// Data type for collecting informations about the streamer
+class StreamerStatusType {
+  friend class StdIOWriter;
+  friend class JSONWriter;
+  friend class FlatbuffersWriter;
+
+public:
+  StreamerStatusType();
+  StreamerStatusType(const StreamerStatusType &other);
+  StreamerStatusType(StreamerStatusType &&other) noexcept = default;
+
+  ~StreamerStatusType() = default;
+
+  StreamerStatusType &operator=(const StreamerStatusType &other);
+  StreamerStatusType &operator=(StreamerStatusType &&other);
+  StreamerStatusType &operator+=(const StreamerStatusType &other);
+  StreamerStatusType &operator-=(const StreamerStatusType &other);
+  StreamerStatusType operator+(const StreamerStatusType &other);
+
+  void reset();
+
+  double bytes{0.0};
+  double messages{0.0};
+  double errors{0.0};
+  double bytes2{0.0};
+  double messages2{0.0};
+};
+
+// Data type for extracting statistics about the streamer
+struct StreamerStatisticsType {
+  double size_avg{0}, size_std{0};
+  double freq_avg{0}, freq_std{0};
+  int partitions{0};
```

Comment:
The number of partition  per topic was an initial request from Dominik. At the moment is not implemented so I will remove the variable, eventually to be restored later.
 :white_check_mark: